### PR TITLE
refactor(multiple): update to typescript 3.7.5 move to dist instead o…

### PR DESCRIPTION
**Pull request by anthonyvercolano**
…f lib parameter checking and suppression

Start changing files to build with 3.7.5 change the transpiling to output to dist, more checking (and suppressing) parameter usage.

Still quite a few directories to go

### CODE REVIEW

File: common/transport/amqp/.npmignore [hunk 1]
```diff
@@ -3,4 +3,5 @@ devdoc/
 test/
 samples/
 src/
-lib/**/*.map
\ No newline at end of file
+dist/**/*.map
+.nyc_output
```
Comment (YoDaMa) [inline common/transport/amqp/.npmignore:7]: is this supposed to be added?

File: common/transport/amqp/src/amqp.ts [hunk 2]
```diff
@@ -213,11 +214,16 @@ export class Amqp {
             this._fsm.transition('connecting', connectionParameters, connectCallback);
           },
           disconnect: (callback) => callback(null, new results.Disconnected()),
+          // @ts-ignore TS6133 endpoint and linkOptions are not used
```
Comment (YoDaMa) [inline common/transport/amqp/src/amqp.ts:217]: would be in favor of not having these and instead using _

File: common/transport/amqp/src/sender_link.ts [hunk 3]
```diff
@@ -118,6 +118,7 @@ export class SenderLink extends EventEmitter implements AmqpLink {
       this._fsm.handle('senderRejectedEvent', context);
     };
 
+    // @ts-ignore TS6133 context not used
```
Comment (YoDaMa) [inline common/transport/amqp/src/sender_link.ts:121]: everywhere.

File: device/samples/simple_sample_device.js [hunk 4]
```diff
@@ -61,13 +57,13 @@ function connectCallback () {
...
 
-};
+}
 
 // fromConnectionString must specify a transport constructor, coming from any transport package.
 let client = Client.fromConnectionString(deviceConnectionString, Protocol);
 
 client.on('connect', connectCallback);
-client.on('error', errorCallback);
+client.on('error', (err) => console.error(err.message));
```
Comment (YoDaMa) [inline device/samples/simple_sample_device.js:66]: could you remove this change.

File: device/transport/http/src/http.ts [hunk 5]
```diff
@@ -68,8 +68,8 @@ export class Http extends EventEmitter implements DeviceTransport {
   private _http: Base;
   private _opts: HttpReceiverOptions;
   private _cronObj: any;
-  private _intervalObj: number;
-  private _timeoutObj: number;
+  private _intervalObj: NodeJS.Timer;
```
Comment (YoDaMa) [inline device/transport/http/src/http.ts:71]: are these necessary for the Typescript upgrade? If not, I'd actually prefer them in their own PR.

Comment (anthonyvercolano) [reply device/transport/http/src/http.ts:71]: Yup, kinda are.  It started beefing about this.

File: common/transport/amqp/src/sender_link.ts (hunk 3 above)
Comment (anthonyvercolano) [reply common/transport/amqp/src/sender_link.ts:121]: Sure.

File: common/transport/amqp/src/amqp.ts (hunk 2 above)
Comment (anthonyvercolano) [reply common/transport/amqp/src/amqp.ts:217]: Great idea.  Thank you.

File: common/transport/amqp/.npmignore (hunk 1 above)
Comment (anthonyvercolano) [reply common/transport/amqp/.npmignore:7]: It certainly seemed like the right think to do.  Consistency between all the directories.

File: device/samples/simple_sample_device.js (hunk 4 above)
Comment (anthonyvercolano) [reply device/samples/simple_sample_device.js:66]: Sure.

**Pull request merged**